### PR TITLE
Add reload data button

### LINKED ISSUE #163: Working with WFS layers
When working with WFS layers, QGIS and QField do not refresh data from server when navigating. For QGIS, simply clicking on the refresh button allows to refresh layer from server. Wouldn't it be usefull to integrate a refresh button in QField ? Or a way to refresh the map without closing and reopening the project ?

### LINKED ISSUE #665: Reload WFS feature from webservice on form opening and after edit
Reload WFS feature from webservice on form opening and after edit

### Reproduction Steps:

1. Add a new feature on a WFS layer, edit a few attribute, save and close the form in Qfield in a client that we will call "client 1"
2. Click on the feature on the map again in order to open the form => the unique id is = 0 and not the value that is stored in db.
3. Edit the feature attribute from another client (Geomapfish, other QField, ...) "client 2"
4. Click on the feature again => the edits done with "client 2" are not loaded in "client 2"

### Expected behavior:

Edits from "client 2" should be visible when form in opened again in "client 1"

### Observed behavior:

Edits from "client 2" are not visible when form in opened again in "client 1"

**QField version:** master

https://github.com/opengisch/QField/commit/79b14a6845df8406357b81f57732d0333c4edb4c

## Additional information:

Without adressing all the challenges of concurrent edition, it would be a useful improvement to have the attributes of the selected feature reloaded at form opening time. Futher improvement would be an button to reload a given layer.

### PR DESCRIPTION
**Pull request by m-kuhn**
Needs a new sdk to be tested

Fixes #163 
Fixes #665

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 1c97cee44a55794b2e9334822ea8afe5fe482528:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1473-caebafa3eb0652158b1f906e53b8124be21af25d-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1473-caebafa3eb0652158b1f906e53b8124be21af25d-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1473-caebafa3eb0652158b1f906e53b8124be21af25d-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1473-caebafa3eb0652158b1f906e53b8124be21af25d-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 43a98ed7b6b7c2ce48f5d2b92bb9c1ebd890524d:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1473-8a818415136200213e9b96e72c12a8774ad195ff-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1473-8a818415136200213e9b96e72c12a8774ad195ff-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1473-8a818415136200213e9b96e72c12a8774ad195ff-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1473-8a818415136200213e9b96e72c12a8774ad195ff-x86.apk)

File: src/qml/LayerTreeItemProperties.qml [hunk 1]
```diff
@@ -44,6 +46,7 @@ Popup {
     expandCheckBox.text = layerTree.data( index, FlatLayerTreeModel.Type ) === 'group' ? qsTr('Expand group') : qsTr('Expand legend item')
     expandCheckBox.checked = !layerTree.data(index, FlatLayerTreeModel.IsCollapsed)
 
+    reloadDataButtonVisible = true// layerTree.data(index, FlatLayerTreeModel.CanReloadData)
```
Comment (signedav) [inline src/qml/LayerTreeItemProperties.qml:49]: You want to make it depending on `CanReloadData`, no?

**Review by signedav — CHANGES_REQUESTED** [review]

File: src/qml/LayerTreeItemProperties.qml (hunk 1 above)
Comment (m-kuhn) [reply src/qml/LayerTreeItemProperties.qml:49]: right, that is a debug leftover, thanks !

Comment (m-kuhn) [reply src/qml/LayerTreeItemProperties.qml:49]:
```suggestion
    reloadDataButtonVisible = layerTree.data(index, FlatLayerTreeModel.CanReloadData)
```

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for c9b61edca67f976f976b9bb7ab94c7103433704c:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1473-07d1777bda875bb8779f53f98ce3b73ce920d8ba-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1473-07d1777bda875bb8779f53f98ce3b73ce920d8ba-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1473-07d1777bda875bb8779f53f98ce3b73ce920d8ba-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1473-07d1777bda875bb8779f53f98ce3b73ce920d8ba-x86.apk)

**Review by signedav — APPROVED** [review]

**Comment by signedav** [comment]
Thanks :tada:

**Pull request merged**